File: Swift/BOJ/2000/2230.swift
//
//  2230.swift
//  Algorithm
//
//  Created by 유정주 on 2023/06/08.
//

import Foundation

let nm = readLine()!.split { $0 == " " }.map { Int($0)! }
let n = nm[0], m = nm[1]
var nums: [Int] = []
for _ in 0..<n {
    let input = Int(readLine()!)!
    nums.append(input)
}

nums.sort()

var left = 0, right = 1
var result = Int.max
while left <= right && right < n {
    let sub = nums[right] - nums[left]
    if sub == m {
        result = sub
        break
    } else if sub > m {
        result = min(result, sub)
        left += 1
    } else if sub < m {
        right += 1
    }
}

print(result)

File: Python/BOJ/투포인터/2230.py
//
//  2230.py
//  SwiftAlgorithm
//
//  Created by 유정주 on 2023/06/08.
//

import sys; readLine = sys.stdin.readline

n, m = map(int, readLine().split())
nums = []
for _ in range(n):
  num = int(readLine().rstrip())
  nums.append(num)

nums.sort()

result = 2000000000
left = 0; right = 1
while left <= right and right < n:
  sub = nums[right] - nums[left]
  if sub == m:
    result = sub
    break
  elif sub > m:
    result = min(result, sub)
    left += 1
  elif sub < m:
    right += 1

print(result)
